apps/examples/wifi_manager_sample/wm_test_daemon.c

```diff
@@ -0,0 +1,245 @@
+		wifi_scan_iter = wifi_scan_iter->next;
+	}
+	WM_TEST_SIGNAL;
+}
+
+static void wm_get_apinfo(wifi_manager_ap_config_s *apconfig)
+{
+	strncpy(apconfig->ssid, WM_AP_SSID, strlen(WM_AP_SSID) + 1);
+	apconfig->ssid_length = strlen(WM_AP_SSID);
+	apconfig->ap_auth_type = WM_AP_AUTH;
+	if (WM_AP_AUTH != WIFI_MANAGER_AUTH_OPEN) {
+		strncpy(apconfig->passphrase, WM_AP_PASSWORD, strlen(WM_AP_PASSWORD) + 1);
+		apconfig->passphrase_length = strlen(WM_AP_PASSWORD);
+		apconfig->ap_crypto_type = WM_AP_CRYPTO;
+	}
+}
+
+static void wm_get_softapinfo(wifi_manager_softap_config_s *ap_config)
+{
+	strncpy(ap_config->ssid, WM_SOFTAP_SSID, strlen(WM_SOFTAP_SSID) + 1);
+	strncpy(ap_config->passphrase, WM_SOFTAP_PASSWORD, strlen(WM_SOFTAP_PASSWORD) + 1);
+	ap_config->channel = WM_SOFTAP_CHANNEL;
+}
+
+static int run_procedure(void *arg)
+{
+	int nres = 0;
+	wifi_manager_result_e res = WIFI_MANAGER_SUCCESS;
+
+	/* Join to softAP*/
+	WT_LOG(WT_TAG, "connect AP");
+	wifi_manager_ap_config_s apconfig;
+	wm_get_apinfo(&apconfig);
+	res = wifi_manager_connect_ap(&apconfig);
+	if (res != WIFI_MANAGER_SUCCESS) {
+		WT_LOGE(WT_TAG, "connect AP fail %d", res);
+		return -1;
+	}
+	WT_LOG(WT_TAG, "wait connect success event");
+	WM_TEST_WAIT;
+
+	/* Receive data */
+	WT_LOG(WT_TAG, "receive dummy data %d", WT_DATA_SIZE);
+	nres = wt_receive_dummy(WT_DATA_SIZE);
+	if (nres < 0) {
+		WT_LOGE(WT_TAG, "receive data fail %d", nres);
+		return -1;
+	}
+
+	/* Set softAP */
+	WT_LOG(WT_TAG, "start softAP");
+	wifi_manager_softap_config_s softap_config;
+	wm_get_softapinfo(&softap_config);
+	res = wifi_manager_set_mode(SOFTAP_MODE, &softap_config);
+	if (res != WIFI_MANAGER_SUCCESS) {
+		WT_LOGE(WT_TAG, "set softap fail %d", res);
+		return -1;
+	}
+
+	/* Wait join event */
+	WT_LOG(WT_TAG, "wait join event");
+	WM_TEST_WAIT;
+
+	/*  receive data */
+	WT_LOG(WT_TAG, "receive data size %d", WT_DATA_SIZE);
+	nres = wt_receive_dummy(WT_DATA_SIZE);
+	if (nres < 0) {
+		WT_LOGE(WT_TAG, "receive data fail %d", nres);
+		return -1;
+	}
+
+	/*  Wait leave event */
+	WT_LOG(WT_TAG, "wait leave event");
+	WM_TEST_WAIT;
+
+	/* Start STA */
+	WT_LOG(WT_TAG, "start STA");
+	res = wifi_manager_set_mode(STA_MODE, NULL);
+	if (res != WIFI_MANAGER_SUCCESS) {
+		WT_LOGE(WT_TAG, "start STA fail %d", res);
+		return -1;
+	}
+
+	return 0;
+}
+
+static void *run_wm_daemon(void *arg)
+{
+	wifi_manager_result_e wres = WIFI_MANAGER_SUCCESS;
+	int cnt = 0;
+
+	WT_LOG(WT_TAG, "initialize wi-fi");
+	wres = wifi_manager_init(&g_wifi_callbacks);
+	if (wres != WIFI_MANAGER_SUCCESS) {
+		WT_LOGE(WT_TAG, "init fail");
+		return NULL;
+	}
+
+	while (1) {
+		WT_LOG(WT_TAG, "=== test count %d===", cnt);
+		if (run_procedure(arg) < 0) {
+			WT_LOGE(WT_TAG, "error in main task");
+			break;
+		}
+	}
+
+	WT_LOG(WT_TAG, "deinit wi-fi\n");
+	wres = wifi_manager_deinit();
+	if (wres != WIFI_MANAGER_SUCCESS) {
+		WT_LOGE(WT_TAG, "deinit fail");
+		return NULL;
+	}
+
+	return NULL;
```

Comment:
This function always returns NULL.
Is this intended? Or you want to return `wres` for success case?